docs/modules/ROOT/pages/cops_style.adoc

```diff
@@ -7233,6 +7233,34 @@ ENV.fetch(:key, VALUE)
 
 * https://github.com/JuanitoFatas/fast-ruby#hashfetch-with-argument-vs-hashfetch--block-code
 
+== Style/RedundantFileExtensionInRequire
+
+|===
+| Enabled by default | Safe | Supports autocorrection | VersionAdded | VersionChanged
+
+| Pending
+| Yes
+| Yes
+| 0.88
+| -
+|===
+
+This cop checks for the presence of superfluous `.rb` extension in
```

Comment:
You might also add here some of the details about the different type of files `requires` tries and why this is safe here. 
We should probably add a similar guideline to the style guide.